db/machine.go

```diff
@@ -97,9 +338,84 @@ func (d DB) GetMachines() ([]Machine, error) {
 	return machines, rows.Err()
 }
 
+//FILTER LIST OF MACHINE: BY RSERVATION DATES
+func(d DB) Filter_M_By_Dates(from time.Time, to time.Time) ([]Machine, error){
```

Comment:
Ah: here's another filtering method. Please see the general comments on the sorting methods... you should find a way of reducing duplication here.